File: Greedy/Medium/LET_no2279_Maximum Bags With Full Capacity of Rocks.cpp
class Solution {
public:
    int maximumBags(vector<int>& capacity, vector<int>& rocks, int additionalRocks) {
        for (int i = 0; i < capacity.size(); i++)
            capacity[i] -= rocks[i];
        
        sort(capacity.begin(), capacity.end());
        int res = 0;
        for (int num : capacity) {
            if (num > additionalRocks)
                break;
            additionalRocks -= num;
            if (additionalRocks >= 0)
                res++;
        }

        return res;
    }
};

File: Greedy/Medium/LET_no2279_Maximum Bags With Full Capacity of Rocks.py
class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], additionalRocks: int) -> int:
        for i in range(len(capacity)):
            capacity[i] -= rocks[i]
        
        capacity.sort()
        res = 0
        for num in capacity:
            additionalRocks -= num
            if additionalRocks >= 0:
                res += 1
        return res